test/test_gridfs.py

```diff
@@ -540,7 +540,7 @@ def test_gridfs_secondary_lazy(self):
 
         # Connects, doesn't create index.
         self.assertRaises(NoFile, fs.get_last_version)
-        self.assertRaises(NotPrimaryError, fs.put, "data")
+        self.assertRaises(NotPrimaryError, fs.put, "data", encoding="utf-8")
```

Comment:
Why was this needed?

Reply:
I think the try-finally statement originally in `GridFS.put` suppressed an error that should have shown up without the `encoding` kwarg there. `GridFS.put` relies on `GridIn.write`, the latter of which requires that an encoding is set for `str` objects, otherwise it will raise an exception.

Reply:
Can you try removing this change and share the traceback of the test failure (if it fails)?

Reply:
Yeah, this is the one test that fails:

```
======================================================================
ERROR: test_gridfs_secondary_lazy (test.test_gridfs.TestGridfsReplicaSet)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/benwarner/Documents/pymongo/gridfs/grid_file.py", line 350, in write
    read = data.read
AttributeError: 'str' object has no attribute 'read'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/benwarner/Documents/pymongo/gridfs/grid_file.py", line 357, in write
    data = data.encode(self.encoding)
  File "/Users/benwarner/Documents/pymongo/gridfs/grid_file.py", line 252, in __getattr__
    raise AttributeError("GridIn object has no attribute '%s'" % name)
AttributeError: GridIn object has no attribute 'encoding'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/benwarner/Documents/pymongo/test/test_gridfs.py", line 543, in test_gridfs_secondary_lazy
    self.assertRaises(NotPrimaryError, fs.put, "data")
  File "/Users/benwarner/opt/anaconda3/lib/python3.9/unittest/case.py", line 739, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/Users/benwarner/opt/anaconda3/lib/python3.9/unittest/case.py", line 201, in handle
    callable_obj(*args, **kwargs)
  File "/Users/benwarner/Documents/pymongo/gridfs/__init__.py", line 136, in put
    grid_file.write(data)
  File "/Users/benwarner/Documents/pymongo/gridfs/grid_file.py", line 359, in write
    raise TypeError("must specify an encoding for file in order to write str")
TypeError: must specify an encoding for file in order to write str
```

Reply:
Oooh I get it now.